spark-commons/src/test/scala/za/co/absa/spark/commons/errorhandling/implementations/ErrorMessageArrayTest.scala

```diff
@@ -199,4 +194,34 @@ class ErrorMessageArrayTest extends AnyFunSuite with SparkTestBase {
     assert(result == expected)
 
   }
+
+  test("test return errCol type schema") {
+    val errorMessageArray = ErrorMessageArray("errCol")
+
+    val expectedResults = StructType(Seq(
+      StructField("errType", StringType, nullable = false),
+      StructField("errCode", LongType, nullable = false),
+      StructField("errMsg", StringType, nullable = false),
+      StructField("errColsAndValues", MapType(StringType, StringType, valueContainsNull = true), nullable = false),
+      StructField("additionInfo", StringType, nullable = true)
+    ))
+
+    val results = errorMessageArray.errorColumnType
+    assert(results == expectedResults)
+  }
+
+  test("errorColumnAggregationType should return an ArrayType structure for column added during the aggregation") {
+    val errorMessageArray = ErrorMessageArray("errCol")
+
+    val expectedResults = Some(ArrayType(
```

Comment:
Use the same logic here as in `ErrorHandlingFilterRowsWithErrorsTest` with the only difference that instead of `.headOption` use the field identified by its name ("_errCol_") here.